src/main/kotlin/exercise3/task2/ParenthesesClusters.kt

```diff
@@ -24,7 +26,31 @@ package exercise3.task2
  */
 
 internal fun String.splitToBracketsClusters(): List<String> {
-    TODO("Implement me!!!")
+
+    if (this.length %2 == 1) return listOf()
+
+    val open = listOf('(', '[', '{')
+    val stack = Stack<Char>()
+
+    var tempString = ""
+    val returnList = mutableListOf<String>()
+
+    for (i in this){
+        if(i in open){
+            stack.push(i)
+            tempString += i
+        }
+        else {
+            if (stack.isEmpty()) return listOf()
+            tempString += i
+            stack.pop()
+        }
+        if (stack.isEmpty()){
```

Comment:
Can this be done as a last statement in the body of ' else`? 

There is no need to check if the stack is empty when the current character is an open bracket. We know that the `stack.isEmpty()` expression will always return `false` because we pushed the open bracket character to the stack before evaluating it.